proposals/TP1-xds-transport-next.md

```diff
@@ -0,0 +1,1028 @@
+    performant and requires minimal management server state tracking and round
+    trips.
+*   Medium sized collection of objects; list collections with reference entries.
+    The management server only needs to handle subscribe/unsubscribe state
+    tracking, since both the collection and its constituent resources are
+    independent xDS resources.
+*   For any other use case with a large collection of objects and/or highly
+    dynamic updates, glob collections are recommended. The management server needs to
+    provide full delta diffs, since there is no explicit collection object to
+    inform the client when resources come and go.
+
+### Transport network addressing
+
+An `xdstp://` URI specifies a logic authority and it’s necessary to map to some physical 
+network address during the resource fetch process. In the existing xDS configuration
+fetch configuration, this capability is provided by a `ConfigSource` delivered alongside a
+resource name in the xDS resource graph.
+
+This proposal adds to `ConfigSource` the ability to specify the
+authorities for which the `ConfigSource` may be used:
+
+```
+message Authority {
+  string name = 1;
+
+  … when signing is added, items such as CA trust chain, cert pinning … 
+}
+
+message ConfigSource {
+  repeated Authority authorities = N;
+
+  …  extant ConfigSource … 
+}
+```
+
+When the `ConfigSource` adjacent to a resource name matches the URI authority,
+the `ConfigSource` will continue to be used.
+
+When there is no match, the xDS client will fallback to the bootstrap to map
+from authority to `ConfigSource.
+
+
+To support ADS with multiple control planes, the
+[`ApiType`](https://github.com/envoyproxy/envoy/blob/31128e7dc22355876020188bc8feb99304663041/api/envoy/config/core/v3/config_source.proto#L44)
+enum in `ApiConfigSource` will be augmented with `AGGREGATED_DELTA_GRPC`.
+This will replace and deprecate the existing `ads` field in `ConfigSource`. When
+used in `xdstp://` authority resolution, an `ConfigSource` will act as an ADS
+transport if configured with this new enum value. Instead of pointing to a
+single shared ADS `ConfigSource` declared in the bootstrap, any resource
+authority (regardless of type) that maps to a `ConfigSource` with
+`AGGREGATED_DELTA_GRPC` will be multiplexed on the xDS stream specified by the
+`ConfigSource`.
+
+### Discovery request and responses
+
+State-of-the-world (SotW) `Discovery{Request,Response}` are not supported with
+the new naming scheme and are effectively deprecated for servers that make use
+of the scheme. The use of inlined entries in list collections allows simple
+management servers that do not wish to perform state tracking to perform
+state-of-the-world updates. This is achieved by sending the entire state, e.g.
+every listener inlined in a `ListenerCollection`.
+
+In both the xDS transport and resources, existing fields for managing resource
+names will continue to be used. A client and server will detect `xdstp://`
```

Comment:
do we consider to add a client feature capability explicitly for client to tell management server if it supports the xdstp naming scheme?

Reply:
The client basically announces this through its use of `xdstp://` names. It probably would only be useful to do this in `Node` if we do it, since if yo're using `xdstp://` you don't need to announce you're using it :D Would this be useful to control plane implementors from your perspective?

Reply:
Yeah, I don't see how a client capability will help here.  The client has to send the resource name it wants in the same message as it sends its client capabilities, so by the time the server sees the capability, it's already too late.

I think the way that people will use this is simply to not configure clients to use xdstp names unless they're talking to a control plane that supports those names.  After all, just like an HTTP server, it's the xDS server that determines which names match to which contents; the client has to ask for the name that the server is using for whatever content it wants.